### PR TITLE
Added a utility to reduce number of replicas for an offline table

### PR DESCRIPTION
**Pull request by mcvsubbu**
If we want to reduce the number of replicas for an offline table, then
running helix AutoRebalanceStragy to recompute partitions does not work.
It does work if we want to increase number of replicas.

Added a separate utility command to reduce the number of replicas.

Also fixed the Rebalancer to account for IdealState changes during the time
th command is run.

### CODE REVIEW

File: pinot-tools/src/main/java/com/linkedin/pinot/tools/PinotNumReplicaChanger.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import com.google.common.base.Function;
+import com.linkedin.pinot.common.config.AbstractTableConfig;
+import com.linkedin.pinot.common.config.TableNameBuilder;
+import com.linkedin.pinot.common.metadata.ZKMetadataProvider;
+import com.linkedin.pinot.common.utils.helix.HelixHelper;
+import com.linkedin.pinot.common.utils.retry.RetryPolicies;
+import javax.annotation.Nullable;
+
+
+public class PinotNumReplicaChanger extends PinotSegmentRebalancer {
```
Comment (kishoreg) [inline pinot-tools/src/main/java/com/linkedin/pinot/tools/PinotNumReplicaChanger.java]: why is this extending segmentrebalancer?

Comment (mcvsubbu) [inline pinot-tools/src/main/java/com/linkedin/pinot/tools/PinotNumReplicaChanger.java]: Many common methods.
Also, for increasing the number of replicas, we just need to call the rebalancer, I believe. Untested, so I did not check that in.

Comment (kishoreg) [inline pinot-tools/src/main/java/com/linkedin/pinot/tools/PinotNumReplicaChanger.java]: I think i figured that it's re-using lot of methods from segmentrebalancer. This is fine for now but eventually, we need to extract them into separate class.

Comment (mcvsubbu) [inline pinot-tools/src/main/java/com/linkedin/pinot/tools/PinotNumReplicaChanger.java]: Re-factored both classes to inherit from a common base class.

**Pull request merged**